File: app/src/main/java/com/adeleke/samad/eddandega/MainActivity.java
package com.adeleke.samad.eddandega;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;

import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.DatePicker;
import android.widget.ProgressBar;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.fragment.app.FragmentTransaction;
import androidx.viewpager.widget.ViewPager;

import com.adeleke.samad.eddandega.Contract.ContractClass;
import com.adeleke.samad.eddandega.Contract.ContractInterface;
import com.adeleke.samad.eddandega.Fragments.AnthropometryFragment;
import com.adeleke.samad.eddandega.Fragments.FluidFragment;
import com.adeleke.samad.eddandega.Fragments.LMPFragment;
import com.adeleke.samad.eddandega.Presenters.LMPPresenter;
import com.google.android.material.badge.BadgeDrawable;
import com.google.android.material.tabs.TabLayout;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private Toolbar toolbar;
    private TabLayout tabLayout;
    private ViewPager viewPager;

    private LMPFragment lmpFragment;
    private AnthropometryFragment anthropometryFragment;
    private FluidFragment fluidFragment;
    private SharedPreferences sharedPreferences;
    private SharedPreferences.Editor editor;
    private Boolean nightModeIsOn;





@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

         toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        tabLayout = findViewById(R.id.tab_layout);
        viewPager = findViewById(R.id.view_pager);

        lmpFragment = new LMPFragment();
        anthropometryFragment = new AnthropometryFragment();
        fluidFragment = new FluidFragment();

        sharedPreferences = getSharedPreferences("com.adeleke.samad.eddandega", Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
        nightModeIsOn = sharedPreferences.getBoolean("nightMode", false);

        if(nightModeIsOn){
            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
        }else{
            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
        }

        tabLayout.setupWithViewPager(viewPager);
        ViewPagerAdaper viewPagerAdapter = new ViewPagerAdaper(getSupportFragmentManager(), 0);



        viewPagerAdapter.addFragment(lmpFragment, "EDD");
        viewPagerAdapter.addFragment(anthropometryFragment, "Weight");
        viewPagerAdapter.addFragment(fluidFragment, "Fluid");

        viewPager.setAdapter(viewPagerAdapter);

        tabLayout.getTabAt(0).setIcon(R.drawable.ic_pregnant_woman);
        tabLayout.getTabAt(1).setIcon(R.drawable.ic_child);
        tabLayout.getTabAt(2).setIcon(R.drawable.ic_water_drop);

//    BadgeDrawable badgeDrawable = tabLayout.getTabAt(0).getOrCreateBadge();
//    badgeDrawable.setVisible(true)
//    badgeDrawable.setNumber(2);



        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                int position  =tab.getPosition();
                switch (position) {
                    case 0:
                        setupLMPFrag();
                    break;
                    case 1:
                        setUoFluidFrag();
                    break;
                    case 2:
                        setUpAnthropometryFrag();
                }

            }
            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });

        if (savedInstanceState == null){
            setupLMPFrag();
        }

}




    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.toolbar_menu, menu);
        return true;


    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

       if (item.getItemId() == R.id.toggle_night){
          if (nightModeIsOn){
              AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
              nightModeIsOn = false;

          }else{
              AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
              nightModeIsOn = true;
          }
           editor.putBoolean("nightMode", nightModeIsOn);
           editor.apply();
           return super.onOptionsItemSelected(item);
        }
       return false;
    }


    public void setupLMPFrag(){
        LMPFragment lmpFragment = new LMPFragment();
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        //fragmentTransaction.setCustomAnimations(R.anim.enter_from_right, R.anim.exit_to_right, R.anim.enter_from_right, R.anim.exit_to_right);
        //fragmentTransaction.addToBackStack(null);
        fragmentTransaction.add(R.id.fragment_container, lmpFragment, "MESSAGE_FRAG");
        fragmentTransaction.replace(R.id.fragment_container,lmpFragment).commit();
    }


    public void setUoFluidFrag(){
        FluidFragment fluidFragment = new FluidFragment();
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        //fragmentTransaction.setCustomAnimations(R.anim.enter_from_right, R.anim.exit_to_right, R.anim.enter_from_right, R.anim.exit_to_right);
        //fragmentTransaction.addToBackStack(null);
        fragmentTransaction.add(R.id.fragment_container, fluidFragment, "FLUID_FRAG");
        fragmentTransaction.replace(R.id.fragment_container,fluidFragment).commit();
    }

    public void setUpAnthropometryFrag(){
        AnthropometryFragment anthropometryFragment = new AnthropometryFragment();
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        //fragmentTransaction.setCustomAnimations(R.anim.enter_from_right, R.anim.exit_to_right, R.anim.enter_from_right, R.anim.exit_to_right);
        //fragmentTransaction.addToBackStack(null);
        fragmentTransaction.add(R.id.fragment_container, anthropometryFragment, "FLUID_FRAG");
        fragmentTransaction.replace(R.id.fragment_container,anthropometryFragment).commit();
    }


    private class ViewPagerAdaper extends FragmentPagerAdapter {

        private List<Fragment> fragments = new ArrayList<>();
        private List<String> fragmentTitle = new ArrayList<>();

        public ViewPagerAdaper(@NonNull FragmentManager fm, int behavior) {
            super(fm, behavior);
        }

        @NonNull
        @Override
        public Fragment getItem(int position) {
            return fragments.get(position);
        }
        public void addFragment(Fragment fragment, String title){
            fragments.add(fragment);
            fragmentTitle.add(title);
        }

        @Override
        public int getCount() {
            return fragments.size();
        }

        @Nullable
        @Override
        public CharSequence getPageTitle(int position) {
            return fragmentTitle.get(position);
        }
    }
}

File: app/src/main/java/com/adeleke/samad/eddandega/MainActivity.kt
package com.adeleke.samad.eddandega

import android.content.SharedPreferences
import android.content.SharedPreferences.Editor
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.adeleke.samad.eddandega.anthro.AnthroFragment
import com.adeleke.samad.eddandega.databinding.ActivityMainBinding
import com.adeleke.samad.eddandega.fluid.FluidFragment
import com.adeleke.samad.eddandega.lmp.LMPFragment
import java.util.*

class MainActivity : AppCompatActivity() {

    private var lmpFragment: LMPFragment? = null
    private var weightFragment: AnthroFragment? = null
    private var fluidFragment: FluidFragment? = null

    private lateinit var sharedPreferences: SharedPreferences
    private lateinit var editor: Editor

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)

        // Have Access to the toolbar
        setSupportActionBar(binding.toolbar)

        // Initialize the fragments
        lmpFragment = LMPFragment()
        weightFragment = AnthroFragment()
        fluidFragment = FluidFragment()


        sharedPreferences = getSharedPreferences("com.adeleke.samad.eddandega", MODE_PRIVATE)

        when (sharedPreferences.getInt("night", 0)) {
            0 -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
            1 -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
            2 -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
        }

        // Configure tab layout and viewPager
        binding.tabLayout.setupWithViewPager(binding.viewPager)
        val viewPagerAdapter = ViewPagerAdapter(supportFragmentManager, 0)
        viewPagerAdapter.addFragment(lmpFragment!!, "EDD")
        viewPagerAdapter.addFragment(weightFragment!!, "Weight")
        viewPagerAdapter.addFragment(fluidFragment!!, "Fluid")
        binding.viewPager.adapter = viewPagerAdapter
        binding.tabLayout.getTabAt(0)!!.setIcon(R.drawable.ic_pregnant_woman)
        binding.tabLayout.getTabAt(1)!!.setIcon(R.drawable.ic_child)
        binding.tabLayout.getTabAt(2)!!.setIcon(R.drawable.ic_water_drop)

        setContentView(binding.root)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.toolbar_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        var value = 0
        when (item.itemId) {
            R.id.night_on -> {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
                value = 1
            }
            R.id.night_off -> {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
                value = 2
            }
            R.id.night_default -> {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
                value = 0
            }
        }
        editor = sharedPreferences.edit();
        editor.putInt("night", value);
        editor.apply()
        return false
    }

    private inner class ViewPagerAdapter(fm: FragmentManager, behavior: Int) : FragmentPagerAdapter(fm, behavior) {
        private val fragments: MutableList<Fragment> = ArrayList()
        private val fragmentTitle: MutableList<String> = ArrayList()
        override fun getItem(position: Int): Fragment {
            return fragments[position]
        }

        fun addFragment(fragment: Fragment, title: String) {
            fragments.add(fragment)
            fragmentTitle.add(title)
        }

        override fun getCount(): Int {
            return fragments.size
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return fragmentTitle[position]
        }
    }

}